### ISSUE TITLE
Use a ResponseFactory for UserStatus

### ISSUE THREAD

**Issue by jpdillingham** [jpdillingham/Soulseek.NET#599]
Presently user status changed events use type `UserStatusChangedEventArgs`, which is redundant due to the existence of the `UserStatus` type.  Refactor the events to use `UserStatus`, and refactor `UserStatusResponse` into `UserStatusResponseFactory` using existing patterns for similar types.

It's been a while since I've thought about this, but to clarify when and why the `ResponseFactory` pattern is used; `*Response` and `*Notification` types are internal to Soulseek.NET, and therefore can't be exposed via method return types or event handlers.  The `*EventArgs` classes are used as the public model for this data, except in cases where there is a corresponding public method on `ISoulseekClient` used to request the data (as opposed to an unsolicited event).  It doesn't make sense to return an `EventArgs` subclass from a method call, so concrete types were created.  Where a concrete type exists, events that deal with that data should use that concrete type instead of an `EventArgs` subclass.

To illustrate:

* `ServerInfo` is a concrete type that encapsulates data sent by the server
* `ServerInfoReceived` is an event that advises subscribers that data has changed
* `ISoulseekClient` has property `ServerInfo` that makes this data available to callers

The `ServerInfo` property shouldn't return an `EventArgs` subclass, so the `ServerInfo` type is needed.  Because the `ServerInfo` type exists, and a `ServerInfoReceivedEventArgs` class would be identical to `ServerInfo`, the `ServerInfoReceived` event uses `ServerInfo` as event arguments.

To summarize, the general rule is that a public, non-EventArgs type should be used for event arguments, if a suitable type is available.  Otherwise, a new `*EventArgs` class should be used.  `*ResponseFactory` classes are used in these cases to turn generic messages into instances of these public types, instead of the internal `*Response` classes typically used.

Because `UserStatus` is a public, non-EventArgs type, the corresponding event should use `UserStatus` instead of `UserStatusChangedEventArgs`.

Hopefully this note prevents me from having to reason through this again in the future :)
